packit_service/service/events.py

```diff
@@ -405,6 +405,9 @@ def get_package_config(self) -> Optional[PackageConfig]:
             f"\tpr_id: {self.pr_id}"
         )
 
+        if self.commit_sha is not None and self.commit_sha.startswith("0000000"):
+            return None
```

Comment:
Some comment would be nice to express that `00000...` means branch deletion.

Reply:
Could the `deleted` key of the event checked instead, and if it's `True`, simply not create anything in the Celery Queue?

This would be a more straightforward approach, than aborting while getting the package config.

Reply:
This PR solves a slightly different issue:

- The skipping of push that deletes branch is already implemented. (#904 )
- In my refactor, I started adding package-config to the result object which results in errors for this situation. (This pr solves that problem.)

But we can sure do the filtering of events in a different way.

Reply:
Hm, I see. From what I can see #904 didn't solve the original issue (#739) as there were Sentry events for this problem during the weekend (I've linked them to #739).

My point is: I see this filtering happening at this level of the architecture a little bit too late. It's not a "natural" place to do it. I would not expect such filtering to happen neither in `process_jobs()` nor in `get_package_config()`.

Relying on a special value ("00000") to tell that this is branch deletion is also not the best possible solution. There is an explicit field for this in the event payload. We should rather rely on that.

Creating a task for the service to process, when it's obvious at the time of receiving the event payload that this is not going to be needed is problematic, and leads to issues similar to the one this PR tries to solve.

Reply:
> Hm, I see. From what I can see #904 didn't solve the original issue (#739) as there were Sentry events for this problem during the weekend (I've linked them to #739).

It did solve it, but I broke it again by adding package-config to result..;)  (It causes loading config from unknown commit again.)

> My point is: ....

I don't argue. I only wanted to point out that we are speaking about something slightly different.

And, yes, we can implement your proposal. Here or as a follow-up.